### ISSUE TITLE
Remove experimental draft ES constructs

### ISSUE THREAD

**Issue by wyuenho** [cloudflare/cf-ui#22]
We are using a lot of experimental stuff like class properties, a withdrawn syntax for `export default` and object spread. Since only object spread seems to be going somewhere, I propose we remove class-properties and export-extensions. This will also stop ESLint from complaining without having to do weird things to it.

**Comment by thejameskyle** [comment]
Class properties is a pretty safe proposal at this point, it's stage 2 and the only debate is over if `=` should be changed to something like `:=` which we'll probably discuss at this week's tc39 meeting.

Also, dropping `babel-preset-cf` is definitely not what you should be doing. Keep control over your own preset and version it yourself.

**Comment by wyuenho** [comment]
@terinjokes said static properties should be `{configurable: false, enumerable: true, writable: true}` but babel currently sets them all to true. I'd really like to minimize semantic differences. There's enough of that in the ES2015 preset against the real spec as is. We'll discuss more about class properties internally, decisions haven't been made yet.

**Comment by thejameskyle** [comment]
Then turn on `spec` mode for https://github.com/cloudflare/babel-preset-cf/blob/master/index.js#L7 if you're that concerned about it.

```diff
-     require("babel-plugin-transform-class-properties"),
+     [require("babel-plugin-transform-class-properties"), { spec: true }],
```

**Comment by wyuenho** [comment]
I just read the TC39 minutes and the issues in the repo. It seems like there are still talks about what `this`, `super`, `arguments` should be bound to, interactions with decorators, private fields and inheritance. There's also computed properties to consider. There's too much still up in the air now.

Separately, `babel-present-cf` isn't being used much both external and internally. There are a number of problems with this preset. The `loose` options being one, incomplete ES2015 syntax and experimental syntax being another. The current preset makes it easy to introduce stable syntax but difficult to convert unstable or non-spec-compliant syntax to spec. It's far easier to remove dependents on it now then later. We can come up with an updated company-side application preset and library preset later if we desire.

**Comment by thejameskyle** [comment]
Silly

**Comment by wyuenho** [comment]
:)

**Comment by wyuenho** [comment]
Fixed in PR#25
